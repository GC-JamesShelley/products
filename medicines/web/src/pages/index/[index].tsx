import React, { useEffect } from 'react';
import { NextPage, NextPageContext } from 'next';
import { useRouter } from 'next/router';

import DrugIndex from '../../components/drug-index/index';
import Page from '../../components/page';
import SearchWrapper from '../../components/search-wrapper';
import { useLocalStorage } from '../../hooks';
import substanceLoader from '../../services/substance-loader';
import { ISubstance } from '../../model/substance';

interface IAppProps {
  results: ISubstance[];
  substanceIndex: string;
}

const App: NextPage<IAppProps> = props => {
  const [storageAllowed, setStorageAllowed] = useLocalStorage(
    'allowStorage',
    false,
  );
  const router = useRouter();

  useEffect(() => {
    if (!props.substanceIndex) {
      router.push('/');
    }
  }, [props.substanceIndex]);

  useEffect(() => {
    if (window) {
      window.scrollTo(0, 0);
    }
  }, []);

  return (
    <Page
      title="Products"
      storageAllowed={storageAllowed}
      setStorageAllowed={setStorageAllowed}
    >
      <SearchWrapper initialSearchValue="">
        <DrugIndex
          title={`${props.substanceIndex || '...'}`}
          items={props.results}
        />
      </SearchWrapper>
    </Page>
  );
};

App.getInitialProps = async (context: NextPageContext): Promise<IAppProps> => {
  const {
    query: { index },
  } = context;
  let results = [];
  let substanceIndex = '';

  if (index) {
    substanceIndex = index.toString();
    results = await substanceLoader.load(substanceIndex);
  }

  return {
    results,
    substanceIndex,
  };
};

export default App;
