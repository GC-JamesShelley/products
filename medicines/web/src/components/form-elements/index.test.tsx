import { shallow } from 'enzyme';
import React from 'react';
import { Button, Checkbox } from './index';

describe('Button', () => {
  it('should render', () => {
    const component = shallow(<Button type="submit" value="val" />);
    expect(component).toMatchSnapshot();
  });
});

describe('Checkbox', () => {
  it('should render', () => {
    const changeHandler = () => {
      return null;
    };
    const component = shallow(
      <Checkbox
        value="val"
        name="agree"
        id="agree-checkbox"
        onChange={changeHandler}
      />,
    );
    expect(component).toMatchSnapshot();
  });
});
