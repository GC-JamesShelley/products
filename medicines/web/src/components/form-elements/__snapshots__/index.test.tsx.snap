// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
<input
  className="sc-AxirZ fdlzyX"
  type="submit"
  value="val"
/>
`;

exports[`Checkbox should render 1`] = `
<styled.input
  id="agree-checkbox"
  name="agree"
  onChange={[Function]}
  type="checkbox"
  value="val"
/>
`;
